src/modules/connection.js

```diff
@@ -55,28 +57,38 @@ class Connection {
    * Since RabbitMQ advise to open one channel by process and node is mono-core, we keep only 1 channel for the whole connection.
    * @return {Promise} A promise that resolve with an amqp.node channel object
   */
-  getChannel() {
+  getChannel(queue, prefetch) {
     const url = this._config.host;
-    const { prefetch } = this._config;
     const connection = this.connections[url];
 
+    queue = queue || defaultChannel;
+    const channelToUse = prefetch ? queue : defaultChannel;
+
     // cache handling, if channel already opened, return it
-    if (connection && connection.chann) {
-      return Promise.resolve(connection.chann);
+    if (connection && connection.channels[channelToUse]) {
+      if (prefetch) {
+        throw new Error(`Channel already exist for queue ${queue}`);
+      }
+      return Promise.resolve(connection.channels[channelToUse]);
+    }
+
+    const configWithPrefetch = Object.assign({}, this._config);
+    if (prefetch) {
+      configWithPrefetch.prefetch = prefetch;
     }
 
-    connection.chann = connection.conn.createChannel()
+    connection.channels[channelToUse] = connection.conn.createChannel()
       .then((channel) => {
-        channel.prefetch(prefetch);
+        channel.prefetch(configWithPrefetch);
```

Comment:
This doesn't seem correct, you pass the value of prefetch and not the whole config